### DISCUSSION TITLE
[PhpUnitBridge] Parallel testing - coverage

### DISCUSSION THREAD

**Discussion by jgrygierek** [symfony/symfony#43549] [Q&A]
Hi, I'm using PHPUnit Bridge to test my code. My tests are divided into few parts (smoke, unit, functional). Each of them runs parallelly. But I don't know how to count coverage in this case? It is easy if I run all tests at once, so I have only one file and I can use it. But in my case I need one coverage file for each part. How I can do that?

**Comment by jgrygierek** [comment]
Any help?

**Comment by lyrixx** [comment]
Hello,

Since you are running tests in parallel, you'll have to merge the coverage yourself.

If you managed to do so (I should not be that hard), you may open a PR to Symfony do that everyone could benefit your works.

Thanks

**Reply by jgrygierek** [reply]
But I need to have multiple files in this case. The problem is that phpunit bridge runs it in parallel, creating only one file, which is overwritten by every new test type.

**Reply by lyrixx** [reply]
Oups, that's problematic indeed! Do you want to work on a solution?

**Reply by jgrygierek** [reply]
I don't have time to do it, sorry. At least not now.

**Reply by lyrixx** [reply]
No problem 👍🏼 If you got the answer to your question, don't forget to mark this discussion as answer
